packages/mui-material/src/PigmentHidden/PigmentHidden.tsx

```diff
@@ -0,0 +1,173 @@
+'use client';
+import * as React from 'react';
+import clsx from 'clsx';
+import { Breakpoint } from '@mui/system';
+// @ts-ignore
+import Hidden from '@pigment-css/react/Hidden';
+import capitalize from '@mui/utils/capitalize';
+import composeClasses from '@mui/utils/composeClasses';
+import HiddenJs from '../Hidden/HiddenJs';
+import { getHiddenCssUtilityClass } from '../Hidden/hiddenCssClasses';
+import { useTheme } from '../zero-styled';
+
+export interface HiddenProps {
+  /**
+   * The content of the component.
+   */
+  children?: React.ReactNode;
+  /**
+   * Specify which implementation to use.  'js' is the default, 'css' works better for
+   * server-side rendering.
+   * @default 'js'
+   */
+  implementation?: 'js' | 'css';
+  /**
+   * You can use this prop when choosing the `js` implementation with server-side rendering.
+   *
+   * As `window.innerWidth` is unavailable on the server,
+   * we default to rendering an empty component during the first mount.
+   * You might want to use a heuristic to approximate
+   * the screen width of the client browser screen width.
+   *
+   * For instance, you could be using the user-agent or the client-hints.
+   * https://caniuse.com/#search=client%20hint
+   */
+  initialWidth?: Breakpoint;
+  /**
+   * If `true`, component is hidden on screens below (but not including) this size.
+   * @default false
+   */
+  lgDown?: boolean;
```

Comment:
While we are working on finalizing the Grid API, I think we should also update this component to follow a similar API.

Reply:
Hidden is a bit complicated as it contains `HiddenJs` implementation. I don't think changing the API would be simple and what's should be the name of the prop.

How do we handle this?
```js
<Hidden smDown lgUp>
```

Reply:
The major issue I see currently is the props (TS or prop-types). It's hardcoded to the default breakpoints. But if someone changes that, they won't get correct autocompletion. This is also an issue for core Material UI packages.

Reply:
> The major issue I see currently is the props (TS or prop-types)

You mean just for this component or the layout components?

Reply:
For all the layout components that we have that have similar props that depend on the theme.